### PR TITLE
 [safari] Fix authentication

### PR DESCRIPTION
**Pull request by phan-ctrl**
Fix issue #22161

## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fix issue #22161 because missing cookies.

### CODE REVIEW

File: youtube_dl/extractor/safari.py [hunk 1]
```diff
@@ -71,6 +71,10 @@ def is_logged(urlh):
         # oreilly serves two same groot_sessionid cookies in Set-Cookie header
         # and expects first one to be actually set
         self._apply_first_set_cookie_header(urlh, 'groot_sessionid')
+        # Fix issue #22161
+        # We need 2 more cookies to be able to login to oreilly
+        self._apply_first_set_cookie_header(urlh, 'orm-jwt')
+        self._apply_first_set_cookie_header(urlh, 'orm-rt')
```
Comment (dstftw) [inline youtube_dl/extractor/safari.py]: DRY.

Comment (phan-ctrl) [reply youtube_dl/extractor/safari.py]: @dstftw updated

File: youtube_dl/extractor/safari.py [hunk 2]
```diff
@@ -70,7 +70,11 @@ def is_logged(urlh):
 
         # oreilly serves two same groot_sessionid cookies in Set-Cookie header
         # and expects first one to be actually set
-        self._apply_first_set_cookie_header(urlh, 'groot_sessionid')
+        # Fix issue #22161
+        # We need 2 more cookies to be able to login to oreilly
+        cookies = ['groot_sessionid', 'orm-jwt', 'orm-rt']
```
Comment (dstftw) [inline youtube_dl/extractor/safari.py]: Inline.

Comment (phan-ctrl) [reply youtube_dl/extractor/safari.py]: done

**Pull request merged**